src/core/pointcloud/qgspointcloudrenderer.cpp

```diff
@@ -190,4 +195,71 @@ void QgsPointCloudRenderer::setPointSymbol( PointSymbol symbol )
 }
 
+QVector<QMap<QString, QVariant>> QgsPointCloudRenderer::identify( QgsPointCloudLayer *layer, QgsRenderContext renderContext, const QgsGeometry &geometry )
+{
+  QVector<QMap<QString, QVariant>> selectedPoints;
+
+  QgsPointCloudIndex *index = layer->dataProvider()->index();
+  const IndexedPointCloudNode root = index->root();
+
+  const float maxErrorPixels = renderContext.convertToPainterUnits( maximumScreenError(), maximumScreenErrorUnit() );// in pixels
+
+  QgsRectangle rootNodeExtentMapCoords = index->nodeMapExtent( root );
+  try
+  {
+    rootNodeExtentMapCoords = renderContext.coordinateTransform().transformBoundingBox( index->nodeMapExtent( root ) );
+  }
+  catch ( QgsCsException & )
+  {
+    QgsDebugMsg( QStringLiteral( "Could not transform node extent to map CRS" ) );
+  }
+
+  const float rootErrorInMapCoordinates = rootNodeExtentMapCoords.width() / index->span(); // in map coords
+
+  double mapUnitsPerPixel = renderContext.mapToPixel().mapUnitsPerPixel();
+  if ( ( rootErrorInMapCoordinates < 0.0 ) || ( mapUnitsPerPixel < 0.0 ) || ( maxErrorPixels < 0.0 ) )
+  {
+    QgsDebugMsg( QStringLiteral( "invalid screen error" ) );
+    return selectedPoints;
+  }
+
+  float maxErrorInMapCoordinates = maxErrorPixels * mapUnitsPerPixel;
+
+  QgsGeometry selectionGeometry = geometry;
+  if ( geometry.type() == QgsWkbTypes::PointGeometry )
+  {
+    double x = geometry.asPoint().x();
+    double y = geometry.asPoint().y();
+    if ( pointSymbol() == QgsPointCloudRenderer::PointSymbol::Square )
+    {
+      QgsPointXY deviceCoords = renderContext.mapToPixel().transform( QgsPointXY( x, y ) );
+      QgsPointXY point1( deviceCoords.x() - 2 * pointSize(), deviceCoords.y() - 2 * pointSize() );
+      QgsPointXY point2( deviceCoords.x() + 2 * pointSize(), deviceCoords.y() + 2 * pointSize() );
+      QgsPointXY point1MapCoords = renderContext.mapToPixel().toMapCoordinates( point1.x(), point1.y() );
+      QgsPointXY point2MapCoords = renderContext.mapToPixel().toMapCoordinates( point2.x(), point2.y() );
+      QgsRectangle pointRect( point1MapCoords, point2MapCoords );
+      selectionGeometry = QgsGeometry::fromRect( pointRect );
+    }
+    else if ( pointSymbol() == QgsPointCloudRenderer::PointSymbol::Circle )
+    {
+      QgsPoint centerMapCoords( x, y );
+      QgsPointXY deviceCoords = renderContext.mapToPixel().transform( centerMapCoords );
+      QgsPoint point1( deviceCoords.x(), deviceCoords.y() - 2 * pointSize() );
+      QgsPoint point2( deviceCoords.x(), deviceCoords.y() + 2 * pointSize() );
+      QgsPointXY point1MapCoords = renderContext.mapToPixel().toMapCoordinates( point1.x(), point1.y() );
+      QgsPointXY point2MapCoords = renderContext.mapToPixel().toMapCoordinates( point2.x(), point2.y() );
+      QgsCircle circle = QgsCircle::from2Points( QgsPoint( point1MapCoords ), QgsPoint( point2MapCoords ) );
+      QgsPolygon *polygon = circle.toPolygon( 6 );
+      QgsGeometry circleGeometry( polygon );
```

Comment:
```suggestion
      std::unique_ptr< QgsPolygon > polygon( circle.toPolygon( 6 ) );
      QgsGeometry circleGeometry( std::move( polygon ) );
```

(avoids the raw pointer, which makes me nervous to look at :smile: )